main.c

```diff
@@ -139,7 +160,14 @@ static void cmd_exec(char *param) {
 		swayidle_log_errno(LOG_ERROR, "fork failed");
 	} else {
 		swayidle_log(LOG_DEBUG, "Spawned process %s", param);
-		waitpid(pid, NULL, 0);
+		if (state.wait) {
+			swayidle_log(LOG_DEBUG, "Blocking until process exits");
+		}
+		int status;
+		waitpid(pid, &status, 0);
+		if (state.wait) {
```

Comment:
Okay, one last comment: can we guard this `if` with `WIFEXITED`, so that we don't print garbage when the process has been signaled? Also can we zero-initialize `status` so that it doesn't have undefined contents even if `waitpid` fails?

Let me know if you're tired of the endless review, I can do the final touches.

Reply:
I do not mind either way - I am grateful that you have taken the time to guide me through this. That being said, it does sound like it would be quicker for you to do the final touches so please do so if it doesn't burden you too much.

Reply:
Ah, no, if you don't feel burnt out please do the changes -- by doing it that way contributors can learn more :P

Reply:
OK I will aim for sometime this week :+1:

Reply:
Done in the last force push